examples/quickstart/event_synchronization.cpp

```diff
@@ -38,26 +38,26 @@ struct data
 };
 
 ///////////////////////////////////////////////////////////////////////////////
-void worker(std::size_t i, data& d, pika::lcos::local::counting_semaphore& sem)
+void worker(std::size_t i, data& d, pika::counting_semaphore<>& sem)
 {
     d.init.wait();
     std::cout << d.msg << ": " << i << "\n" << std::flush;
-    sem.signal();    // signal main thread
+    sem.release();    // signal main thread
```

Comment:
I think you could use a `latch` here instead, similarly to how it's used in `enumerate_threads.cpp` (except I think you can `latch::count_down` here, instead of latch::arrive_and_wait`).